From af747e69eb7ed5e469346f8ac2a4a0930a28d2d8 Mon Sep 17 00:00:00 2001
From: Albin B <albin.bala.krishnan@intel.com>
Date: Fri, 22 Jan 2016 16:26:32 +0530
Subject: [PATCH 12/13] [EM/USBC_PD]: Fix issue of boot with apple dock
 connected vconn state

When booting with apple dock, vconn state shown none, since hard reset
is triggered from sink. Hence moving to vconn sink state after
completion of hard reset during sink state.

Change-Id: I928681c89d9ba7d969ab9e45ffe59a1621b53e1b
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-17170
Signed-off-by: Albin B <albin.bala.krishnan@intel.com>
Reviewed-on: https://android.intel.com:443/461341
---
 drivers/usb/typec/pd/policy_engine.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/usb/typec/pd/policy_engine.c b/drivers/usb/typec/pd/policy_engine.c
index 81562d8..8fd3ae6 100644
--- a/drivers/usb/typec/pd/policy_engine.c
+++ b/drivers/usb/typec/pd/policy_engine.c
@@ -134,11 +134,13 @@ static void pe_do_dpm_reset_complete(struct policy_engine *pe)
 	if (pe->cur_prole == POWER_ROLE_SOURCE) {
 		/* VBUS On if source*/
 		devpolicy_set_vbus_state(pe->p.dpm, true);
-		/*VCONN on if source*/
+		/* VCONN on if source*/
 		devpolicy_set_vconn_state(pe->p.dpm, VCONN_SOURCE);
 		/* Reset data role to DFP*/
 		pe_set_data_role(pe, DATA_ROLE_DFP);
 	} else if (pe->cur_prole == POWER_ROLE_SINK) {
+		/* VCONN off if sink */
+		devpolicy_set_vconn_state(pe->p.dpm, VCONN_SINK);
 		/* Reset data role to UFP*/
 		pe_set_data_role(pe, DATA_ROLE_UFP);
 	} else
-- 
1.9.1

